main.tex

```diff
@@ -432,6 +436,54 @@ \section{Using Domain-Specific Languages to improve the
 without building an expensive prototype, and advanced types of hosting language
 encoding initial system specification provide verified-by-construction result. 
 
+\section{Tool support}
+
+Formalising the design specifications is a good approach to the development of a resilient
+system. Specifications, represented via mathematical models, can indeed be verified and
+simulated, targeting an error-free system. Workcraft provides the support for different
+models, this can help the design for resiliency. Conditional partial order graphs (CPOG)
+(available in the Workcraft framework), for instance, can support the development of
+processor instruction sets. In Figure \ref{fig:screenshot}, all the steps from the
+specification of each instruction, to the synthesis of the final microcontroller are
+depicted. Instructions can be specified in the form of algebraic equations at first, then the
+graphs can be imported, encoded and synthesised in the form of a microcontroller. The latter
+can be mapped with the usage of a technology library and eventually simulated: internal and
+external signals can be visualised via digital waveforms, where dependencies among signals
+are shown. The development of a domain specific language (DSL), that can be used for the
+specification of the instructions, still remains a gap of this flow. Having a scalable, yet
+reusable language would be beneficial for resiliency. The instruction correctness, indeed,
+could be simulated and formally verified, the partial order should be derived from the
+specifications.
+
+\begin{figure*}[ht!]
+\begin{center}
+	\includegraphics[width=\linewidth]{FIG/screen.png}
+	\caption{From the specification of the partial orders, to the synthesis and simulation
+	of the microcontroller. Screenshot from Workcraft.}
+	\label{fig:screenshot}
+\end{center}
+\end{figure*}
+
+The CPOG and Dataflow Structures (DFS) were used together to design an asynchronous
+reconfigurable pipeline for the computation of the ordinal pattern encoding. CPOG was used
+for the design of the control unit, which is used for the reconfigurability of the pipeline
+(3-18 stages). DFS supported the design of the datapath asynchronous components instead. Each
+module has indeed been modelled and simulated in Workcraft at first, and then translated into
+hardware description (VHDL). The DFS model matches the HDL description, the asynchronous 4-
```

Comment:
I translated the model in VHDL manually. I would turn VHDL into Verilog, since Workcraft can generate verilog files.